src/xacro/__init__.py

```diff
@@ -413,18 +413,27 @@ def process_include(elt, macros, symbols, func):
         except TypeError:
             raise XacroException('namespaces are supported with --inorder option only')
 
-    for filename in get_include_files(filename_spec, symbols):
-        # extend filestack
-        oldstack = push_file(filename)
-        include = parse(None, filename).documentElement
+    optional = True if optional == "True" else False
 
-        # recursive call to func
-        func(include, macros, symbols)
-        included.append(include)
-        import_xml_namespaces(elt.parentNode, include.attributes)
+    try:
```

Comment:
Do you really want to have the try outside of the for loop?
Consider the case of multiple include files specified with globbing (`*.xacro`). In this case, on a failure, all remaining include files will be skipped as well. Having the try inside the for loop, you can just continue processing the remaining files.

Reply:
Done